src/wallet/wallet.cpp

```diff
@@ -493,6 +495,49 @@ void CWallet::AddToSpends(const uint256& wtxid)
         AddToSpends(txin.prevout, wtxid);
 }
 
+void CWallet::AddToUnspents(const CWalletTx* wtx, bool filterIsMine)
+{
+    uint256 hash = wtx->GetHash();
+
+    for (unsigned int i = 0; i < wtx->vout.size(); ++i)
+    {
+        const COutPoint outpoint(hash, i);
+        pair<TxSpends::const_iterator, TxSpends::const_iterator> range;
+        range = mapTxSpends.equal_range(outpoint);
+
+        if(range.first == range.second) {
+            isminetype ismine = IsMine(wtx->vout[i]);
+            if(!filterIsMine || ismine) {
+                mapTxUnspents[outpoint] = make_pair(wtx, ismine);
+            }
+        }
+    }
+}
+
+void CWallet::RemoveFromUnspents(const COutPoint &outpoint)
+{
+    UnspentsMap::const_iterator uit = mapTxUnspents.find(outpoint);
+    if(uit != mapTxUnspents.end())
+        mapTxUnspents.erase(uit);
+}
+
+void CWallet::RefreshUnspents()
```

Comment:
@promag it's meant for that, but also to update the isMine value. The thing is, when we load a transaction from the DB, we may not have all the addresses loaded at that point. And we wouldn't know if the address is in our wallet or not. This function goes through all the outputs, after the wallet loads completely.